src/player.ts

```diff
@@ -1,32 +1,79 @@
+export enum PlayerState {
+    ON_GROUND,
+    IN_AIR,
+    LEFT_SLIDE,
+    RIGHT_SLIDE
+}
 export class Player extends Phaser.Physics.Arcade.Sprite {
     jumpCount: number
     body: Phaser.Physics.Arcade.Body
-    inAir: boolean
+    state: PlayerState
+    movingLeft: boolean
+    movingRight: boolean
+    //constants. modify these to tweak 
+    static GRAVITY: number = 50;
+    static X_INC: number = 60;
+    static TOP_SPEED: number = 600;
 
-    constructor(scene: Phaser.Scene,x: number, y: number,  image: string ) {
+    constructor(scene: Phaser.Scene, x: number, y: number, image: string) {
         super(scene, x, y, image)
         this.jumpCount = 2;
         this.body = new Phaser.Physics.Arcade.Body(scene.physics.world, this)
-        this.inAir = false;
+        this.state = PlayerState.IN_AIR;
+        this.movingLeft = false;
+        this.movingRight = false;
     }
 
     jump() {
         if (this.jumpCount) {
             this.setVelocityY(-1000);
             this.jumpCount--;
+            if (this.state == PlayerState.LEFT_SLIDE) {
+                this.setVelocityX(Player.TOP_SPEED);
```

Comment:
Possibly `SLIDING_SPEED`?

Reply:
discussed. this is for bouncing inthe opposite direction